internal/duck_dom/sidebar.go

```diff
@@ -0,0 +1,35 @@
+package duckdom
+
+import "fmt"
+
+type Sidebar struct {
+	ActiveChildId int
+	Children      []Renderable
+	Pos           Position
+	Content       string
+	Styles        string
+}
+
+func (self *Sidebar) OnClick() {}
+
+func (self *Sidebar) Render() string {
+	buffer := ""
```

Comment:
we should also start using string builder for operations like this
for example

```go
...
var builder strings.Builder
...
            builder.WriteString(fmt.Sprintf(MOVE_CURSOR_TO_POSITION, i, j)
            builder.WriteString(DEBUG_STYLES)
            builder.WriteString(" ")
            builder.WriteString(RESET_STYLES)
....
```
also for the sidebar or other big blocks we should try planning strings before building them

set styles
a. move cursor to a line
b. make a row of spaces
reset styles